utils/python/CIME/SystemTests/pem.py

```diff
@@ -12,119 +12,29 @@
 
-class PEM(SystemTestsCommon):
+class PEM(SystemTestsCompareTwo):
 
     def __init__(self, case):
         """
         initialize a test object
         """
-        SystemTestsCommon.__init__(self, case)
-
-    def build_phase(self, sharedlib_only=False, model_only=False):
-        """
-        Build two cases.  Case one uses defaults, case2 uses half the number of threads
-        and tasks. This test will fail for components (e.g. pop) that do not reproduce exactly
-        with different numbers of mpi tasks.
-        """
-        exeroot = self._case.get_value("EXEROOT")
-        cime_model = CIME.utils.get_model()
-
-        machpes1 = os.path.join("LockedFiles","env_mach_pes.PEM1.xml")
-        if ( os.path.isfile(machpes1) ):
-            shutil.copy(machpes1,"env_mach_pes.xml")
-        else:
-            logging.warn("Copying env_mach_pes.xml to %s"%(machpes1))
-            shutil.copy("env_mach_pes.xml", machpes1)
+        SystemTestsCompareTwo.__init__(self, case, True)
 
+    def _common_setup(self):
         self._case.set_value("SMP_BUILD","0")
-        for bld in range(1,3):
-            logging.warn("Starting bld %s"%bld)
-
-            if (bld == 2):
-                # halve the number of tasks
-                for comp in ['ATM','CPL','OCN','WAV','GLC','ICE','ROF','LND']:
-                    ntasks    = self._case.get_value("NTASKS_%s"%comp)
-                    if ( ntasks > 1 ):
-                        self._case.set_value("NTASKS_%s"%comp, ntasks/2)
-
-            self._case.flush()
-            case_setup(self._case, test_mode=True, reset=True)
-            self.clean_build()
-            self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
-            if (not sharedlib_only):
-                shutil.move("%s/%s.exe"%(exeroot,cime_model),
-                            "%s/%s.exe.PEM%s"%(exeroot,cime_model,bld))
-
-            shutil.copy("env_mach_pes.xml", "env_mach_pes.xml.%s"%bld )
-        #
-        # Because mira/cetus interprets its run script differently than
-        # other systems we need to copy the original env_mach_pes.xml# back
-        #
-        shutil.copy(machpes1,"env_mach_pes.xml")
-        shutil.copy("env_mach_pes.xml", os.path.join("LockedFiles","env_mach_pes.xml"))
-
-    def _pem_first_phase(self):
-
-        # Reset beginning test settings
-        expect(os.path.isfile("env_mach_pes.xml.1"),
-               "ERROR: env_mach_pes.xml.1 does not exist, run case.build" )
-
-        shutil.copy("env_mach_pes.xml.1", "env_mach_pes.xml")
-        shutil.copy("env_mach_pes.xml.1", "LockedFiles/env_mach_pes.xml")
-
-        exeroot = self._case.get_value("EXEROOT")
-        cime_model = CIME.utils.get_model()
-        exefile  = "%s/%s.exe"%(exeroot,cime_model)
-        exefile1 = "%s/%s.exe.PEM1"%(exeroot,cime_model)
-        if (os.path.isfile(exefile)):
-            os.remove(exefile)
-        shutil.copy(exefile1, exefile)
-
         self._case.set_value("CONTINUE_RUN",False)
```

Comment:
To be consistent with other (newer) tests, you should move the settings of CONTINUE_RUN and REST_OPTION to config_tests.xml, too. Then - together with my previous comment - you could remove _common_setup entirely.